kubernetes/linux/main.sh

```diff
@@ -1,5 +1,44 @@
 #!/bin/bash
 
+# check to see if Ikey is properly formatted an instrumentation key
+if [[ $APPLICATIONINSIGHTS_AUTH =~ ^[A-Za-z0-9=]+$ ]]  # regex for base64 encoded data
+then
+      # base64 encoded ikey was passed, decode ikey and write to environment variable
+      export APPLICATIONINSIGHTS_AUTH=$(echo $APPLICATIONINSIGHTS_AUTH )
+else
+      # need to fetch ikey from storage account
+      for BACKOFF in {0..5}; do
+            KEY=$(curl -sS $APPLICATIONINSIGHTS_AUTH )
+
+            # there's no easy way to get the HTTP status code from curl, so just check if the result is well formatted
+            if [[ $KEY =~ ^[A-Za-z0-9=]+$ ]]; then
+                  break
+            else
+                  # (exponential backoff)
+                  sleep $((2**$BACKOFF / 4))
+            fi
+      done
+
+      # validate that the retrieved data is an instrumentation key
+      if [[ $KEY =~ ^[A-Za-z0-9=]+$ ]]; then
+            export APPLICATIONINSIGHTS_AUTH=$(echo $KEY)
+            echo "Using cloud-specific instrumentation key"
+      else
+            # no ikey can be retrieved. Put some value in the ikey and disable telemetry
+            export APPLICATIONINSIGHTS_AUTH=NzAwZGM5OGYtYTdhZC00NThkLWI5NWMtMjA3ZjM3NmM3YmRi
+            export DISABLE_TELEMETRY=true
```

Comment:
I didnt see that, you are writing DISABLE_TELEMETRY value to ~/.bashrc. dont we need to write so that we can disable telemetry in case where we failed to get the IKEY?

Reply:
Good point, I'll fix that